prow/pod-utils/downwardapi/jobspec.go

```diff
@@ -46,17 +48,13 @@ type JobSpec struct {
 
 // NewJobSpec converts a kube.ProwJobSpec invocation into a JobSpec
 func NewJobSpec(spec kube.ProwJobSpec, buildID, prowJobID string) JobSpec {
-	refs := kube.Refs{}
-	if spec.Refs != nil {
-		refs = *spec.Refs
-	}
-
 	return JobSpec{
 		Type:      spec.Type,
 		Job:       spec.Job,
 		BuildID:   buildID,
 		ProwJobID: prowJobID,
-		Refs:      refs,
+		Refs:      spec.Refs,
```

Comment:
As long as we update all components at the same time it should be fine, and also non-periodic jobs will still always get a valid ref.

Reply:
I took a glance, looks like the ref is only consumed in sidecar/gcsuploads, and is only checked for presubmit jobs. (I would assume we don't rely on refs for periodic jobs or we were in trouble already :-) )

Reply:
Ah, bootstrap doesn't consume the jobspec env var, it defines its own args that have to be supplied in prowjob config (:-1: ). In that case bootstrap jobs aren't affected by this.

Obviously existing periodic jobs must not be relying on the contents of the `ref` field since it is empty, but existing job wrappers that check for the existence of a ref by checking if `ref.org == "" || ref.repo == ""` would now segfault. I'm not sure if there are any jobs that do this, but something very similar to this caused problems with the job URL template previously so I just wanted to mention it.